modules/user_accounts/php/edit_user.class.inc

```diff
@@ -1287,8 +1287,6 @@ class Edit_User extends \NDB_Form
      */
     private function _getEmailError(\Database $DB, string $email): ?string
     {
-        // remove illegal characters
-        $email = filter_var($email, FILTER_SANITIZE_EMAIL);
```

Comment:
I'm not sure I fully understand the PR description. Is the problem that the email is being filtered, or is the problem that filter_var is being called twice? (Once here, and once in the if statement immediately below).

It's not clear to me from the PHP documentation what filter_var is supposed to remove when there's invalid characters. The documentation says "Returns the filtered data, or false if the filter fails." but playing around with it on the commandline the filter seems to "fail" instead of filter out invalid characters when I add a `>` instead of returning the string with the characters filtered out.

Maybe the PHP documentation is just unclear and it means "return the exact same string or false if it fails the filter", but it seems like it might be safer to keep this line and change the if statement to `if($email === false)`?

Reply:
It's not called twice. Different parameter. One is filter, the other is validate

Reply:
Ok, now I understand. I'll let someone else review it but this makes sense to me.